.github/workflows/release.yaml

```diff
@@ -0,0 +1,100 @@
+name: Release
+on: 
+  workflow_dispatch:
+    inputs:
+      releaseVersion:
+        description: 'Release Version'     
+        required: true
+        default: '0.0.1'
+      snapshotVersion:
+        description: 'Snapshot Version'
+        required: true
+        default: '0.1.0-SNAPSHOT'
+      branch:
+        description: 'Branch to Release'
+        required: false
+        default: 'main'
+permissions:
+  contents: write
+jobs:
+  release:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+        with:
+          ref: ${{ github.event.inputs.branch }}
+      - name: Set up Java
+        uses: actions/setup-java@v4
+        with:
+          java-version: '17'
+          distribution: 'temurin'
+          cache: maven
+      - name: Update Sources
+        run: |
+          mvn versions:set -DgenerateBackupPoms=false -DnewVersion=${{ github.event.inputs.releaseVersion }}
+          mvn -B clean verify
+          sed --in-place --regexp-extended "s|flink-examples-data-generator:([^[:space:]]*)|flink-examples-data-generator:${{ github.event.inputs.releaseVersion }}|g" recommendation-app/data-generator.yaml
+          sed --in-place --regexp-extended "s|flink-examples-data-generator:([^[:space:]]*)|flink-examples-data-generator:${{ github.event.inputs.releaseVersion }}|g" interactive-etl/data-generator.yaml
+          git config --global user.name "${GITHUB_ACTOR}"
+          git config --global user.email "${GITHUB_ACTOR_ID}+${GITHUB_ACTOR}@users.noreply.github.com"
+          git add .
+          git commit -m "Release ${{ github.event.inputs.releaseVersion }}"
+          git tag -m "v${{ github.event.inputs.releaseVersion }}" -a "v${{ github.event.inputs.releaseVersion }}"
```

Comment:
very similar to the later script setting it back into dev mode, should extract a script